### PR TITLE
CMake: Use CMake targets to link against Boost and OpenSSL

### PR DESCRIPTION
**Pull request by neverpanic**
This fixes the build in situations where Boost or OpenSSL headers/libraries are not in the default search path.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -60,7 +60,7 @@ set(LIBRARY_PUBLIC_HEADERS
 add_library(${LIBRARY_NAME} SHARED ${LIBRARY_SOURCES})
 add_library(${PROJECT_NAME}::${LIBRARY_NAME} ALIAS ${LIBRARY_NAME})
 target_include_directories(${LIBRARY_NAME} PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/cwrap")
-target_link_libraries(${LIBRARY_NAME} ${OPENSSL_LIBRARIES})
```
Comment (fetzerch) [inline src/CMakeLists.txt]: You might want to bump the required CMake version to 3.5. (https://gitlab.kitware.com/cmake/cmake/commit/3f9b081f6ee85e0691c36496d989edbe8382589d)

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -60,7 +60,7 @@ set(LIBRARY_PUBLIC_HEADERS
 add_library(${LIBRARY_NAME} SHARED ${LIBRARY_SOURCES})
 add_library(${PROJECT_NAME}::${LIBRARY_NAME} ALIAS ${LIBRARY_NAME})
 target_include_directories(${LIBRARY_NAME} PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/cwrap")
-target_link_libraries(${LIBRARY_NAME} PUBLIC ${OPENSSL_LIBRARIES} Boost::boost)
+target_link_libraries(${LIBRARY_NAME} PUBLIC OpenSSL::Crypto OpenSSL::SSL Boost::boost)
```
Comment (fetzerch) [inline src/CMakeLists.txt]: These will work with CMake > 3.4: https://gitlab.kitware.com/cmake/cmake/commit/75e3a8e811b290cb9921887f2b086377af90880f

Comment (neverpanic) [reply src/CMakeLists.txt]: Done

File: src/CMakeLists.txt (hunk 1 above)
Comment (neverpanic) [reply src/CMakeLists.txt]: Done

**Pull request merged**